src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/QueryTimeoutsTest.java

```diff
@@ -0,0 +1,189 @@
+import com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder;
+import com.feedzai.commons.sql.abstraction.engine.DatabaseEngine;
+import com.feedzai.commons.sql.abstraction.engine.DatabaseEngineDriver;
+import com.feedzai.commons.sql.abstraction.engine.DatabaseEngineException;
+import com.feedzai.commons.sql.abstraction.engine.DatabaseEngineTimeoutException;
+import com.feedzai.commons.sql.abstraction.engine.DatabaseFactory;
+import com.feedzai.commons.sql.abstraction.engine.DatabaseFactoryException;
+import com.feedzai.commons.sql.abstraction.engine.testconfig.DatabaseConfiguration;
+import com.feedzai.commons.sql.abstraction.engine.testconfig.DatabaseTestUtil;
+import com.feedzai.commons.sql.abstraction.entry.EntityEntry;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.ImmutableList;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import static com.feedzai.commons.sql.abstraction.ddl.DbColumnType.INT;
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.column;
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.dbEntity;
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.table;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.ENGINE;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.JDBC;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.SELECT_QUERY_TIMEOUT;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.PASSWORD;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.SCHEMA_POLICY;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.USERNAME;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+
+/**
+ * Unit tests for the query timeout feature. Tests run a query that takes very long to run
+ * with a short timeout and verify that the statement aborts with the right error code.
+ */
+@RunWith(Parameterized.class)
+public class QueryTimeoutsTest {
+
+    /**
+     * Query timeouts supported only on the Postgres, SQL Server and MySQL. SQL Server is not
+     * included in the list below because it it able to handle the test query in ms, and so
+     * is not testeable in this test.
+     */
+    private static List<DatabaseEngineDriver> SUPPORTED_DRIVERS = ImmutableList.of(
+            DatabaseEngineDriver.POSTGRES,
+            DatabaseEngineDriver.MYSQL
+    );
+
+    /**
+     * A query that makes a sort in 1.000.000.000 rows, taking WAY more than
+     * the test timeout until the first row is returned.
+     */
+    private static Query HEAVY_QUERY = SqlBuilder
```

Comment:
suggestion: instead of this query, the some DBs have "sleep" functions that can be used in queries.
What do you think about something like
```
        switch (driver) {
            case POSTGRES:
                heavy = "SELECT pg_sleep(" + QUERY_MINIMUM_TIME + ")";
                break;
            case SQLSERVER:
                heavy = "WAITFOR DELAY '00:00:" + QUERY_MINIMUM_TIME + "'; SELECT 1";
                break;
            case MYSQL:
                heavy = "SELECT SLEEP(" + QUERY_MINIMUM_TIME + ")";
                break;
            default:
                assumeTrue(driver + " engine doesn't support setting timeouts, tests will be skipped", false);
```
where QUERY_MINIMUM_TIME would be the time to sleep in seconds

Reply:
Much better indeed!